sdk/identity/azure-identity/src/managed_identity_source.cpp

```diff
@@ -484,29 +484,19 @@ std::unique_ptr<ManagedIdentitySource> ImdsManagedIdentitySource::Create(
       credName + " will be created" + WithSourceMessage("Azure Instance Metadata Service")
           + ".\nSuccessful creation does not guarantee further successful token retrieval.");
 
-  std::string imdsHost = "http://169.254.169.254";
-  std::string customImdsHost = Environment::GetVariable("AZURE_IMDS_CUSTOM_AUTHORITY_HOST");
-  if (!customImdsHost.empty())
-  {
-    IdentityLog::Write(
-        IdentityLog::Level::Informational, "Custom IMDS host is set to: " + customImdsHost);
-    imdsHost = customImdsHost;
-  }
-  Azure::Core::Url imdsUrl(imdsHost);
-  imdsUrl.AppendPath("/metadata/identity/oauth2/token");
-
   return std::unique_ptr<ManagedIdentitySource>(
-      new ImdsManagedIdentitySource(clientId, objectId, resourceId, imdsUrl, options));
+      new ImdsManagedIdentitySource(clientId, objectId, resourceId, options));
 }
 
 ImdsManagedIdentitySource::ImdsManagedIdentitySource(
     std::string const& clientId,
     std::string const& objectId,
     std::string const& resourceId,
-    Azure::Core::Url const& imdsUrl,
     Azure::Core::Credentials::TokenCredentialOptions const& options)
     : ManagedIdentitySource(clientId, std::string(), options),
-      m_request(Azure::Core::Http::HttpMethod::Get, imdsUrl)
+      m_request(
+          Azure::Core::Http::HttpMethod::Get,
```

Comment:
There is a part of me that isn't happy with magic URLs like this one - this rather critical URL is buried deep in the code of the managed identity source.

How stable over time is this URL? Would it make sense to make it a `constexpr const char*` in a more prominent location?

Reply:
https://learn.microsoft.com/azure/virtual-machines/instance-metadata-service

"IMDS is a REST API that's available at a well-known, non-routable IP address (169.254.169.254). You can only access it from within the VM. Communication between the VM and IMDS never leaves the host."

Reply:
It is very stable (has been the same since the initial impl, and heavily documented). But I agree, we should make it into a named constepxr so the code is self-descriptive.

Reply:
Btw, that kind of information is insanely helpful to have in the documentation (code).